src/firefly/js/ui/tap/TapUtil.js

```diff
@@ -783,6 +783,13 @@ export function getServiceLabel(serviceUrl) {
     return (serviceUrl && (tapOps.find( (e) => e.value===serviceUrl)?.labelOnly)) || '';
 }
 
+export function getServiceId(serviceUrl) {
+    const tapOps= getTapServices();
+    const id= (serviceUrl && (tapOps.find( (e) => e.value===serviceUrl)?.serviceId));
+    if (id) return id;
+    return getServiceLabel(serviceUrl).replaceAll(/\s/g,'');
```

Comment:
Not sure if this logic will cause problem for 'Exoplanet Archive' known service because other than removing space you omitted 'NASA' too?